### PR TITLE
chore(pageserver): remove unnecessary logging in tenant task loops

### LINKED ISSUE #3830: With disabled gc/compaction period logs get polluted
_Originally posted by @knizhnik in https://github.com/neondatabase/neon/pull/3673#discussion_r1129719189_
            
I would just remove the logging in tests, make it more rare or make it once. We very rarely need set period to zero in production, but at the same time, should we forget either to zero in production that would be bad.

Ideas on how to make it more rare or make it once:
- `#[cfg(feature = "testing")]`

Places of logging:
- https://github.com/neondatabase/neon/blob/48936d44f8cc991f2888f580ad7894887b701a75/pageserver/src/tenant/tasks.rs#L109-L111
- https://github.com/neondatabase/neon/blob/48936d44f8cc991f2888f580ad7894887b701a75/pageserver/src/tenant/tasks.rs#L180-L182

### PR DESCRIPTION
**Pull request by duguorong009**
## Problem
- Close #3830 

## Summary of changes
- Add the `#[cfg(feature = "testing")]` attribute to unnecessary loggings in `pageserver/src/tenant/tasks.rs`

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Review by koivunej — APPROVED** [review]

**Comment by koivunej** [comment]
Thanks for working on this! I'll try to get a CI run done for this :)

File: pageserver/src/tenant/tasks.rs [hunk 1]
```diff
@@ -102,6 +102,7 @@ async fn compaction_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
             let started_at = Instant::now();
 
             let sleep_duration = if period == Duration::ZERO {
+                #[cfg(feature = "testing")]
```
Comment (koivunej) [inline pageserver/src/tenant/tasks.rs:105]: Actually I think we need

```suggestion
                #[cfg(not(feature = "testing"))]
```

To have these nagging us in prod, but not while doing (manual) testing.

Comment (koivunej) [reply pageserver/src/tenant/tasks.rs:105]: These could be even refactored to a method similar to `warn_when_period_overrun` with similar or equal to "task" argument, then we'd have a single place to log this "we don't want the repeated log line in tests, but we want it in prod".

Comment (duguorong009) [reply pageserver/src/tenant/tasks.rs]: Thanks for the tip! @koivunej 
I think if there are more cases, we can add helper method similar to `warn_when_period_overrun`.

**Comment by koivunej** [comment]
reset the other PR to [5eebaf9](https://github.com/neondatabase/neon/pull/5188/commits/5eebaf9f846a7ffb42a341e75305ad7e6377125b)

**Comment by duguorong009** [comment]
> reset the other PR to [5eebaf9](https://github.com/neondatabase/neon/pull/5188/commits/5eebaf9f846a7ffb42a341e75305ad7e6377125b)

Um, sorry, what do you mean by `reset the other PR`?

**Comment by koivunej** [comment]
> > reset the other PR to [5eebaf9](https://github.com/neondatabase/neon/pull/5188/commits/5eebaf9f846a7ffb42a341e75305ad7e6377125b)
> 
> Um, sorry, what do you mean by `reset the other PR`? @koivunej Do you mean the other PRs I created? PR #5165 & PR #5196

No, I meant the https://github.com/neondatabase/neon/pull/5193 which I reset to that commit. Just a reminder for myself I don't do it twice :)

I think this is good now, thanks.

Confirmed no more logging, [old](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5198/6095094899/index.html#suites/b97efae3a617afb71cb8142f5afa5224/cd56bda099ba24b4/) vs. [this pr](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5193/6095044597/index.html#suites/b97efae3a617afb71cb8142f5afa5224/dadb8c1b606d544c/).

**Pull request merged**